### PR TITLE
Update cf-rd to draft version 20

### PR DESCRIPTION
**Pull request by pokgak**
This PR updates cf-rd to the latest resource directory draft ([version 20](https://tools.ietf.org/html/draft-ietf-core-resource-directory-20)).

### CODE REVIEW

File: cf-rd/src/main/java/org/eclipse/californium/tools/resources/KeyValuePair.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public String getValue() {
 		return value==null ? "" : value;
 	}
 	
-	public int getIntValue() {
+	public int getIntValue() throws NumberFormatException {
```
Comment (boaks) [inline cf-rd/src/main/java/org/eclipse/californium/tools/resources/KeyValuePair.java]: In "the most" other cases we use the approach to mention `RuntimeException` rather in the javadoc then to throws them explicit.

File: cf-rd/src/main/java/org/eclipse/californium/tools/resources/RDNodeResource.java [hunk 2]
```diff
@@ -317,41 +337,36 @@ private boolean updateEndpointResources(String linkFormat) {
...
 	}
 
-	private void buildLinkFormat(Resource resource, StringBuilder builder, List<String> query) {
+	private List<String> buildLinkFormat(Resource resource, List<String> query) {
+		List<String> resources = new ArrayList<>();
 		if (resource.getChildren().size() > 0) {
 
 			// Loop over all sub-resources
 			for (Resource res : resource.getChildren()) {
+				String link = "";
```
Comment (boaks) [inline cf-rd/src/main/java/org/eclipse/californium/tools/resources/RDNodeResource.java]: I would appreciate, if `StringBuilder` gets used.

**Comment by boaks** [comment]
Thanks! Great work.
I will need some time to try it out. So my current forecast would be, that if you correct the minors above, I merge the PR next Tuesday. Or do you need it faster?

**Comment by boaks** [comment]
In the 2.0.x branch, the `LinkFormat` `COUNT` and `PAGE` seems to be missing.
Do you prepare a additional PR for that to make the changes of this PR working?

File: cf-rd/src/main/java/org/eclipse/californium/tools/resources/KeyValuePair.java (hunk 1 above)
Comment (pokgak) [reply cf-rd/src/main/java/org/eclipse/californium/tools/resources/KeyValuePair.java]: Okay, I'll drop this commit.

File: cf-rd/src/main/java/org/eclipse/californium/tools/resources/RDNodeResource.java (hunk 2 above)
Comment (pokgak) [reply cf-rd/src/main/java/org/eclipse/californium/tools/resources/RDNodeResource.java]: Changed to StringBuilder in fixup commit.

**Comment by pokgak** [comment]
I pushed some fixup commits to address your comments. The check should be all green again after squashing.



> Do you prepare a additional PR for that to make the changes of this PR working?

Opened PR eclipse/californium#964 to address this.

> I will need some time to try it out. So my current forecast would be, that if you correct the minors above, I merge the PR next Tuesday. Or do you need it faster?

Tuesday sounds good. I'm not in a hurry to use this, so you can take your time reviewing :)

**Comment by boaks** [comment]
The "Signed-off-by:"  is missing in the "fixup!" commits.
Please don't waste your time, just squash all your commits to one, sign the result and force push them to your branch.

**Comment by pokgak** [comment]
Squashed. Checks all green. 
Not used to this workflow. I will keep that in mind next time. Thank you!

**Comment by boaks** [comment]
could you please squash all your commits (from 5af644a4cf to d2e67fabafa) into one? 
And then rebase that commit to the current 2.0.x head (we need to fix a executor dependency).
Afterwards "push forced" it to your branch, then I can merge it.
Thanks in advance.

**Comment by pokgak** [comment]
Squashed and rebased.

**Comment by boaks** [comment]
Thanks!

**Pull request merged**